opal/mca/btl/ofi/btl_ofi_context.c

```diff
@@ -144,6 +144,9 @@ mca_btl_ofi_context_t *mca_btl_ofi_context_alloc_scalable(struct fi_info *info,
     mca_btl_ofi_context_t *contexts;
     tx_attr.op_flags = FI_DELIVERY_COMPLETE;
 
+    /* Add RMA/ATOMIC for one-sided to work */
+    tx_attr.caps = (FI_RMA | FI_ATOMIC);
```

Comment:
Related to [libfabric API](https://github.com/ofiwg/libfabric/blob/0c364c0deeeeda6ee2b8eb394f22da16cb3f04c4/man/fi_endpoint.3.md)
> caps - Capabilities
The requested capabilities of the context. The capabilities must be a subset of those requested of the associated endpoint. See the CAPABILITIES section of fi_getinfo(3) for capability details. If the caps field is 0 on input to fi_getinfo(3), the applicable capability bits from the fi_info structure will be used.
The following capabilities apply to the transmit attributes: FI_MSG, FI_RMA, FI_TAGGED, FI_ATOMIC, FI_READ, FI_WRITE, FI_SEND, FI_HMEM, FI_TRIGGER, FI_FENCE, FI_MULTICAST, FI_RMA_PMEM, FI_NAMED_RX_CTX, FI_COLLECTIVE, and FI_XPU.
Many applications will be able to ignore this field and rely solely on the fi_info::caps field. Use of this field provides fine grained control over the transmit capabilities associated with an endpoint. It is useful when handling scalable endpoints, with multiple transmit contexts, for example, and allows configuring a specific transmit context with fewer capabilities than that supported by the endpoint or other transmit contexts.

It reads to me that the intended use case for `fi_tx_attr` is to **scope down** capabilities that were set at the endpoint level. But I'm not sure if the PR is motivated by this.

Otherwise I somehow smell a bug in the libfabric provider that requires this change - why would it require setting `FI_RMA | FI_ATOMIC` again?

Did I miss anything?

Reply:
I agree @wenduwan .  something's off with the provider that needs this change.

Reply:
The CXI provider works fine with mpi-onesided without this change.